User can now set LLDP polling time by REST API #29

cmagnobarbosa opened the pull request:
Today, the user cannot set the LLDP `POLLING_TIME` at run time. This new feature adds a REST terminal, allowing the user change this time. But this modification is not persistent, if NApp restarts, the time set in the `settings.py` file will be loaded and used.

Main Changes:
- Add a REST endpoint that allows the user to update `POLLING_TIME` at the run time.
- Add some uni tests.

Issue fix #33 

Related the discuss about etcd -  [EP020](https://github.com/kytos/kytos/blob/master/docs/blueprints/EP020.rst)

cmagnobarbosa committed:
User can now set LLDP polling time by REST API #29

ajoaoff commented on main.py:
```diff
@@ -369,3 +370,20 @@ def enable_lldp(self):
+        """Get LLDP polling time."""
+        return jsonify({"Polling time in seconds": self.polling_time}), 200
+
+    @rest('v1/time/<time_sec>', methods=['POST'])
```
I think you should not put the value of what you want to change in the URL, but use the payload for that.
I also am not sure if POST is the best method here.

cmagnobarbosa replied:
Thanks for the suggestion @ajoaoff, I changed this method to use the payload. Sorry for the long delay to answer.

hdiogenes commented on main.py:
```diff
@@ -369,3 +370,20 @@ def enable_lldp(self):
+
+    @rest('v1/time/<time_sec>', methods=['POST'])
+    def set_time(self, time_sec):
+        """Set LLDP polling time."""
```
Please add comment or log warning the user that this change will not be saved permanently.

cmagnobarbosa replied:
Thanks for the suggestion @hdiogenes, I added a log warning indicating that this change is not persistent.
``` python
log.info("Polling time has been updated, but this change will not be saved permanently.")
```

hdiogenes commented on main.py:
```diff
@@ -369,3 +370,20 @@ def enable_lldp(self):
         return jsonify({msg_error:
                         error_list}), 400
+
+    @rest('v1/time', methods=['GET'])
```
```suggestion
    @rest('v1/polling_time', methods=['GET'])
```

cmagnobarbosa committed:
Improving the readability of the REST Endpoint

Co-authored-by: Humberto Diógenes <hdiogenes@gmail.com>

cmagnobarbosa committed:
User can now update polling_time at run time by sending a payload

The user can now update polling_time at run time. This commit also adds
some unit tests to verify that these methods work correctly.

hdiogenes commented on main.py:
```diff
@@ -371,3 +372,23 @@ def enable_lldp(self):
+            self.polling_time = int(payload['polling_time'])
+            self.execute_as_loop(self.polling_time)
+            log.info("Polling time has been updated,"
+                     "but this change will not be saved permanently.")
```
Please add the new polling time to the log.

cmagnobarbosa replied:
I added the `polling_time` to log.

rmotitsuki commented:
Maybe set_time should accept only non-zero positive numbers?

rmotitsuki commented:
The unit test **test_set_time** should be splitted in success test and failure test, otherwise if the first assert fail all the other assert will not be evalueted.

cmagnobarbosa committed:
Add LLDP polling time information to the log

The log now contains more information about the LLDP polling time.
The unit test has also fragmented, allowing for more detailed testing.

cmagnobarbosa commented:
Thanks for the suggestion @rmotitsuki, I have divided the unit test and add the abs(), in the `polling_time`.

hdiogenes commented on main.py:
```diff
@@ -371,3 +372,24 @@ def enable_lldp(self):
+
+    @rest('v1/polling_time', methods=['GET'])
+    def get_time(self):
+        """Get LLDP polling time."""
+        return jsonify({"Polling time in seconds": self.polling_time}), 200
```
```suggestion
        """Get LLDP polling time in seconds."""
        return jsonify({"polling_time": self.polling_time}), 200
```

cmagnobarbosa committed:
Now return polling_time in seconds.

Co-authored-by: Humberto Diógenes <hdiogenes@gmail.com>

cmagnobarbosa committed:
Update openapi.yml
